cli/azd/pkg/environment/manager.go

```diff
@@ -221,6 +168,111 @@ func (m *manager) LoadOrCreateInteractive(ctx context.Context, environmentName s
 	return env, nil
 }
 
+func (m *manager) loadOrInitEnvironment(ctx context.Context, environmentName string) (*Environment, bool, error) {
+	// If there's a default environment, use that
+	if environmentName == "" {
+		var err error
+		environmentName, err = m.azdContext.GetDefaultEnvironmentName()
+		if err != nil {
+			return nil, false, fmt.Errorf("getting default environment: %w", err)
+		}
+	}
+
+	if environmentName != "" {
+		env, err := m.Get(ctx, environmentName)
+		switch {
+		case errors.Is(err, ErrNotFound):
+			msg := fmt.Sprintf("Environment '%s' does not exist, would you like to create it?", environmentName)
+			shouldCreate, promptErr := m.console.Confirm(ctx, input.ConsoleOptions{
+				Message:      msg,
+				DefaultValue: true,
+			})
+			if promptErr != nil {
+				return nil, false, fmt.Errorf("prompting to create environment '%s': %w", environmentName, promptErr)
+			}
+			if !shouldCreate {
+				return nil, false, fmt.Errorf("environment '%s' not found: %w", environmentName, err)
+			}
+		case err != nil:
+			return nil, false, fmt.Errorf("loading environment '%s': %w", environmentName, err)
+		case err == nil:
+			return env, false, nil
+		}
+	}
+
+	// Two cases if we get to here:
+	// - The user has not specified an environment name, and there was no default environment set
+	// - The user has specified an environment name, but the named environment didn't exist and they told us they would
+	//   like us to create it.
+	if environmentName != "" && !IsValidEnvironmentName(environmentName) {
+		fmt.Fprintf(
+			m.console.Handles().Stdout,
+			"environment name '%s' is invalid (it should contain only alphanumeric characters and hyphens)\n",
+			environmentName)
+		return nil, false, fmt.Errorf(
+			"environment name '%s' is invalid (it should contain only alphanumeric characters and hyphens)",
+			environmentName)
+	}
+
+	// No environment name, no default environment set.
+	// Ask the user if they want to create a new environment or select an existing one
+	if environmentName == "" {
+		envs, err := m.List(ctx)
+		if err != nil {
+			return nil, false, err
+		}
+
+		choices := make([]string, 0, len(envs)+1)
+		choices = append(choices, "Create a new environment")
+		for _, env := range envs {
+			choices = append(choices, env.Name)
+		}
+
+		sel, err := m.console.Select(ctx, input.ConsoleOptions{
+			Message: "Select an environment to use:",
+			Options: choices,
+		})
+		if err != nil {
+			return nil, false, err
+		}
+
+		if sel > 0 {
+			// Return an existing environment
+			env, err := m.Get(ctx, choices[sel])
+			if err != nil {
+				return nil, false, err
+			}
+
+			setDefault, err := m.console.Confirm(ctx, input.ConsoleOptions{
+				Message:      fmt.Sprintf("Set '%s' as the default environment?", env.Name()),
+				DefaultValue: true,
+			})
```

Comment:
IMHO, we should make this by default and without asking.  I see very very few changes from someone to create a new env and expect it not to become the default selection.  What do you think?

Maybe just a heads-up for the customer like 

```
env foo has been set as default. Use azd env select envName to select a different environment
```

Reply:
This would also match the behavior of `azd env new` which also sets the just created environment as the default.

Reply:
Just to clear the confusion, the prompt here is only issued when selecting an existing environment. However, I think the same logic applies so I removed the defaulting prompt.